graphql/api/src/test/java/com/expediagroup/streamplatform/streamregistry/graphql/mutation/impl/ConsumerBindingMutationImplTest.java

```diff
@@ -100,6 +101,39 @@ public void deleteWithCheckExistDisabledWhenEntityDoesNotExist() {
     assertTrue(result);
   }
 
+  @Test
+  public void updateStatusWithEntityStatusEnabled() {
+    ReflectionTestUtils.setField(consumerBindingMutation, "entityStatusEnabled", true);
+    ConsumerBindingKeyInput key = getConsumerBindingInputKey();
+    Optional<ConsumerBinding> consumerBinding = Optional.of(getConsumer(key));
+    StatusInput statusInput = InputHelper.statusInput();
+
+    when(consumerBindingView.get(any())).thenReturn(consumerBinding);
+    when(consumerBindingService.updateStatus(any(), any())).thenReturn(consumerBinding);
+
+    ConsumerBinding result = consumerBindingMutation.updateStatus(key, statusInput);
+
+    verify(consumerBindingView, times(1)).get(key.asConsumerBindingKey());
+    verify(consumerBindingService, times(1)).updateStatus(consumerBinding.get(), statusInput.asStatus());
+    assertEquals(consumerBinding.get(), result);
+  }
+
+  @Test
+  public void updateStatusWithEntityStatusDisabled() {
+    ReflectionTestUtils.setField(consumerBindingMutation, "entityStatusEnabled", false);
+    ConsumerBindingKeyInput key = getConsumerBindingInputKey();
+    Optional<ConsumerBinding> consumerBinding = Optional.of(getConsumer(key));
+    StatusInput statusInput = InputHelper.statusInput();
+
+    when(consumerBindingView.get(any())).thenReturn(consumerBinding);
+
+    ConsumerBinding result = consumerBindingMutation.updateStatus(key, statusInput);
+
+    verify(consumerBindingView, times(1)).get(key.asConsumerBindingKey());
+    verify(consumerBindingService, never()).updateStatus(consumerBinding.get(), statusInput.asStatus());
+    assertEquals(consumerBinding.get(), result);
+  }
+
   private ConsumerBindingKeyInput getConsumerBindingInputKey() {
```

Comment:
Each entity has these two additional tests to test that when entity status is enabled (default) the entity service is called to update the status. The second test is for when entity status is disabled, the entity service should NOT be called when an updateStatus mutation is received.